archive/main.rs

```diff
@@ -0,0 +1,552 @@
+#[macro_use]
```

Comment:
I would suggest not putting in the `archive/` files here.
If we need them, they'll be in the Git history forever.

Reply:
I agree. I had the same thought and considerated both. I will make another commit and remove archive/